src/ts/state/reducer.ts

```diff
@@ -124,21 +139,41 @@ export interface IPluginState<TMatches extends IMatch = IMatch> {
   };
   // The current error message.
   requestErrors: IMatchRequestError[];
+  // The current state of the filter
+  filterState: TFilterState;
 }
 
 // The transaction meta key that namespaces our actions.
 export const PROSEMIRROR_TYPERIGHTER_ACTION = "PROSEMIRROR_TYPERIGHTER_ACTION";
 
+interface IInitialStateOpts<
+  TFilterState extends unknown,
+  TMatch extends IMatch
+> {
+  doc: Node;
+  matches: TMatch[];
+  ignoreMatch: IIgnoreMatchPredicate;
+  matchColours: IMatchTypeToColourMap;
+  filterOptions: IFilterOptions<TFilterState, TMatch> | undefined;
+}
+
 /**
  * Initial state.
  */
-export const createInitialState = <TMatch extends IMatch>(
-  doc: Node,
-  matches: TMatch[] = [],
-  ignoreMatch: IIgnoreMatch = includeAllMatches,
-  matchColours: IMatchColours = defaultMatchColours
-): IPluginState<TMatch> => {
-  const initialState: IPluginState<TMatch> = {
+export const createInitialState = <
+  TFilterState extends unknown,
+  TMatch extends IMatch
+>({
+  doc,
+  matches = [],
+  ignoreMatch = includeAllMatches,
+  matchColours = defaultMatchColours,
+  filterOptions
```

Comment:
Think this needs `= undefined` for ease of use.

Reply:
See comment above on why this shouldn't be necessary.